examples/d_dmrg.py

```diff
@@ -2,15 +2,22 @@
 
 The example functions in this class do the same as the ones in `toycodes/d_dmrg.py`,
 but make use of the classes defined in tenpy.
+
+.. todo ::
+Set back to using dmrg.run() once that function has been rewritten for Sweep eng.
+Single-site DMRG example.
+Docstrings?
 """
 # Copyright 2018 TeNPy Developers
 
 import numpy as np
 
 from tenpy.networks.mps import MPS
+from tenpy.networks.mpo import MPOEnvironment
 from tenpy.models.tf_ising import TFIChain
 from tenpy.models.spins import SpinModel
 from tenpy.algorithms import dmrg
+from tenpy.algorithms.mps_sweeps import OneSiteH, TwoSiteH
```

Comment:
Because the EffectiveH needs to be called by methods in the `Sweep()` superclass, this seemed to make more sense to put in an `__init__()` method, as is implemented in 1abd1712276773180abe0e006238530c0a6d911c.